src/WebAppDIRAC/Lib/WebHandler.py

```diff
@@ -232,58 +195,43 @@ async def prepare(self):
         We make the assumption that there is always going to be a ``method`` argument
         regardless of the HTTP method used
         """
-
         # Parse request URI
         match = self.PATH_RE.match(self.request.path)
         groups = match.groups()
         self.__setup = groups[0] or Conf.setup()
         self.__group = groups[1]
 
-        # Reset DISET settings
-        self.__disetConfig.reset()
-        self.__disetConfig.setDecorator(self.__disetBlockDecor)
-        self.__disetDump = self.__disetConfig.dump()
-
         try:
             await super().prepare()
         except HTTPError as e:
             raise WErr(e.status_code, e.log_message)
 
-        # Configure DISET with user creds
-        if self.getUserDN():
-            self.__disetConfig.setDN(self.getUserDN())
-        if self.getUserGroup():  # pylint: disable=no-value-for-parameter
-            self.__disetConfig.setGroup(self.getUserGroup())  # pylint: disable=no-value-for-parameter
-        self.__disetConfig.setSetup(self.__setup)
-        self.__disetDump = self.__disetConfig.dump()
-
-        self.__forceRefreshCS()
-
-    def __disetBlockDecor(self, func):
-        def wrapper(*args, **kwargs):
-            raise RuntimeError("All DISET calls must be made from inside a Threaded Task!")
-
-        return wrapper
+    @contextlib.contextmanager
+    def _setupThreadConfig(self):
+        threadConfig = ThreadConfig()
+        if userDN := self.getUserDN():
+            threadConfig.setDN(userDN)
+        if userGroup := self.getUserGroup():
+            threadConfig.setGroup(userGroup)
+        threadConfig.setSetup(self.__setup)
+        try:
+            yield
+        finally:
+            threadConfig.reset()
 
-    def __forceRefreshCS(self):
```

Comment:
I think it is worth adding this deletion to the release notes

Reply:
THis has never been in a release has it and I couldn't find any documentation for it?

Reply:
There is no documentation, this is a specific function for force refresh CS, when adding it an entry was left in release.notes, so deleting it should also be reflected in release.notes (something like "__forceRefreshCS was removed").

---
I do not see the need to leave it as it is, so I agree with its removal. If the force refresh CS request mechanism is needed, I think it should be used at the `BaseRequestHandler` level as the core class.

Reply:
Ah, I see it was added in v4r1p4 but the divergence in the git history made it appear to be something new.

I've updated the release notes to reflect that `X-RefreshConfiguration` is no longer respected as that's the only observable behavour (hopefully nobody is calling `__WebHandler__forceRefreshCS`, which actually might not even be possible to call as it will be resolved to the non-existant `_CLASSNAME__WebHandler__forceRefreshCS`).

Reply:
This is an internal method, it is part of the algorithm that runs during each request, look tornadoes `prepare()` method, Its also described in the `BaseRequestHandler`.

It depended https://github.com/DIRACGrid/DIRAC/pull/4241. In fact, I would not touch it yet and would consider it later, as a less priority.

Reply:
@chrisburr Could you please not remove `__forceRefreshCS` method in this PR, as it is a continuation of the mechanism developed in https://github.com/DIRACGrid/DIRAC/pull/4241. If you think it is necessary to delete it, create a separate PR, so as not to delay the acceptance of this one.

Reply:
so you're right, this logic is gone.. well OK, then I merge this PR